File: collatz.java
import java.util.*;

public class collatz {
	public static void main(String[] args) {
        long num = 1;
        long counterMax = 0;
        while (true)
        {
            long calc = num;
            long counter = 1;
            while (calc != 1)
            {
                if (calc%2 == 0)
                {
                    calc = calc / 2;
                } else {
                    calc = 3*calc + 1;
                }
                counter++;
            }

            if (counterMax < counter)
            {
                counterMax = counter;
                System.out.println("Java Collatz from " + Long.toString(num) + " of len " + Long.toString(counter));
            }
            num++;
        }
	}
}

File: collatz.rb
num = 1
counterMax = 0
while TRUE
    calc = num
    counter = 1
    while calc != 1
        if calc % 2 == 0
            calc = calc.to_i / 2
        else
            calc = 3 * calc.to_i + 1
        counter += 1
        end
    end
    
    if counterMax < counter
        counterMax = counter
        print('Ruby Collatz from ' + num.to_s + " of len " + counter.to_s + "\n")
    end
    num += 1
end